### PR TITLE
Continuous deployment using Github Actions

### PR DESCRIPTION
**Pull request by rasmuscnielsen**
This PR adds

- A new CLI command to generate & encrypt a server private key and install this in the Github repo in a trustless way
- A Github Action job which deploys the server to Vercel with the secret private key

### CODE REVIEW

**Comment by rasmuscnielsen** [comment]
So I just tweaked the implementation now to use environments. This makes it even more secure, as we can completely restrict any workflows to access the secrets unless it is triggered by a commit to their respective branch (staging + main) .

Furthermore we can restrict who can push to main. For now I have added a requirement for only PR that has at least 1 approval. 

I have gone ahead and seeded the Production SERVER_PRIVATE_KEY. 

- Here is a screenshot that validates I used the trustless command (with a dump of the source code): https://drive.google.com/file/d/1HEngcSjhZdDyxc54Qqik-iZMyHgRwH-1/view?usp=sharing
- The corresponding public address is: 0x24F05e73BD7566E2965D50eA0e17F0Bb75BB2a3f

I believe this can be the final public key used for launching the smart contract to production :-)

File: .github/workflows/cicd.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    - name: Commit style fixes
+      uses: stefanzweifel/git-auto-commit-action@v4
+      with:
+        commit_message: Code style fixes [skip ci]
+        commit_user_name: Github Action
+
+  deploy-staging:
+    if: github.ref == 'refs/heads/staging'
+    needs: styling
```
Comment (briangershon) [inline .github/workflows/cicd.yaml:59]: This should be `needs: [tests, styling]` to ensure both run before deploy.

File: .github/workflows/cicd.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      deployOptions: ''
+    secrets:
+      serverPrivateKey: ${{ secrets.SERVER_PRIVATE_KEY }}
+      vercelOrganizationId: ${{ secrets.VERCEL_ORG_ID }}
+      vercelProjectId: ${{ secrets.VERCEL_PROJECT_ID }}
+      vercelToken: ${{ secrets.VERCEL_TOKEN }}
+
+  deploy-production:
+    if: github.ref == 'refs/heads/main'
+    needs: styling
```
Comment (briangershon) [inline .github/workflows/cicd.yaml:72]: Also, this should be `needs: [tests, styling]` to ensure both run before deploy.

**Review by briangershon — APPROVED** [review]
Thanks @rasmuscnielsenI for getting this all running!

I tested by running the scripts locally and looks good! I hadn't heard of `tweetsodium` before, it works well.

I only had two minor feedback comments on `needs` in the github action.

**Comment by rasmuscnielsen** [comment]
Thanks for your feedback @briangershon ! 

I believe `needs: styling` is actually sufficient because the `styling` in turn is set to `needs: tests`. One could of course argue that this dependency is unnecessary, but in its current state I believe its working as expected. 

I will try and merge the PR as is and then we can test & review what happens with the staging deployment and see if we want to tweak that 😊

**Pull request merged**